src/components/UserSelector.tsx

```diff
@@ -1,16 +1,77 @@
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cn from 'classnames';
+import { getUsers } from '../api/users';
+
+type Props = {
+  selectedUser: User | null;
+  onSelectUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectUser,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const handleOutsideClick = useCallback((e: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(e.target as Node)
+    ) {
+      setIsDropdownOpen(false);
+    }
+  }, []);
+
+  const handleUserSelect = useCallback(
+    (user: User) => {
+      onSelectUser(user);
+      setIsDropdownOpen(false);
+    },
+    [onSelectUser],
+  );
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleOutsideClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleOutsideClick);
+    };
+  }, [handleOutsideClick]);
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      ref={dropdownRef}
+      data-cy="UserSelector"
+      className={cn('dropdown', { 'is-active': isDropdownOpen })}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsDropdownOpen(!isDropdownOpen)}
```

Comment:
In this line, you are directly manipulating the DOM state by toggling 'isDropdownOpen'. According to the checklist, it's better to avoid direct DOM manipulation as much as possible and use React instead.